llama_index/indices/multi_modal/base.py

```diff
@@ -0,0 +1,329 @@
+
+    Args:
+        use_async (bool): Whether to use asynchronous calls. Defaults to False.
+        show_progress (bool): Whether to show tqdm progress bars. Defaults to False.
+        store_nodes_override (bool): set to True to always store Node objects in index
+            store and document store even if vector store keeps text. Defaults to False
+    """
+
+    image_store_key = "image"
+
+    def __init__(
+        self,
+        nodes: Optional[Sequence[BaseNode]] = None,
+        index_struct: Optional[IndexDict] = None,
+        service_context: Optional[ServiceContext] = None,
+        storage_context: Optional[StorageContext] = None,
+        use_async: bool = False,
+        store_nodes_override: bool = False,
+        show_progress: bool = False,
+        # Image-related kwargs
+        image_vector_store: Optional[VectorStore] = None,
+        image_embed_model: EmbedType = "clip",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        image_embed_model = resolve_embed_model(image_embed_model)
+        assert isinstance(image_embed_model, MultiModalEmbedding)
+        self._image_embed_model = image_embed_model
+
+        storage_context = storage_context or StorageContext.from_defaults()
+
+        if image_vector_store is not None:
+            storage_context.add_vector_store(image_vector_store, self.image_store_key)
+
+        if self.image_store_key not in storage_context.vector_stores:
+            storage_context.add_vector_store(SimpleVectorStore(), self.image_store_key)
+
+        super().__init__(
+            nodes=nodes,
+            index_struct=index_struct,
+            service_context=service_context,
+            storage_context=storage_context,
+            show_progress=show_progress,
+            use_async=use_async,
+            store_nodes_override=store_nodes_override,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_vector_store(
+        cls,
+        vector_store: VectorStore,
+        service_context: Optional[ServiceContext] = None,
+        # Image-related kwargs
+        image_vector_store: Optional[VectorStore] = None,
+        image_embed_model: EmbedType = "clip",
+        **kwargs: Any,
+    ) -> "VectorStoreIndex":
+        if not vector_store.stores_text:
+            raise ValueError(
+                "Cannot initialize from a vector store that does not store text."
+            )
+
+        storage_context = StorageContext.from_defaults(vector_store=vector_store)
+        return cls(
+            nodes=[],
+            service_context=service_context,
+            storage_context=storage_context,
+            image_vector_store=image_vector_store,
+            image_embed_model=image_embed_model,
+            **kwargs,
+        )
+
+    def as_retriever(self, **kwargs: Any) -> BaseRetriever:
+        raise NotImplementedError("Retriever not yet implemented for MultiModalIndex.")
+
+    def _get_node_with_embedding(
```

Comment:
lol it should be `_get_nodes_with_embeddings` but Vector store base already got this name